api/src/main/resources/client_api.yaml

```diff
@@ -190,6 +184,36 @@ definitions:
         items:
           type: string
 
+  ResultFilters:
+    type: object
+    description: >
+      A list of filters applied to the results of a query. Only results matching the filter criteria
+      should be returned.
+    properties:
+      not:
+        description: >
+         Set to true if a result matching allOf or anyOf below should result in a result *not*
+         being returned.
+      allOf:
```

Comment:
I'm assuming exactly one of these three fields should be populated?

Reply:
Yeah, added a comment.